libcontainer/utils/utils.go

```diff
@@ -36,10 +38,26 @@ func ResolveRootfs(uncleanRootfs string) (string, error) {
 }
 
 // ExitStatus returns the correct exit status for a process based on if it
-// was signaled or exited cleanly.
+// was signaled or exited cleanly
 func ExitStatus(status syscall.WaitStatus) int {
 	if status.Signaled() {
 		return exitSignalOffset + int(status.Signal())
 	}
 	return status.ExitStatus()
 }
+
+// WriteJSON writes the provided struct v to w using standard json marshaling
+func WriteJSON(w io.Writer, v interface{}) error {
+	data, err := json.Marshal(v)
+	if err != nil {
+		return err
+	}
+	n, err := w.Write(data)
+	if err != nil {
+		return err
+	}
+	if l := len(data); int(n) != l {
+		return fmt.Errorf("short write of %d compared to %d", n, l)
```

Comment:
yah thats true ( `Write must return a non-nil error if it returns n < len(p)`), so i dont think we need this check